docs/plugins/common.mdx

```diff
@@ -0,0 +1,709 @@
+| `unsafe_rawColors` | Record<string, `UnsafeRawColor`> | Contains all CSS variables for specific colors with the corresponding function to generate a color based on a given saturation |
+
+### `contextMenu`
+
+Utility module to use together with the [`ContextMenu`](components#contextmenu) component.
+
+:::info
+
+`contextMenu` is still a work in progress.
+
+:::
+
+### `flux`
+
+Flux is an architecture for managing data flow. Stores are objects that can hold state and logic.
+They are responsible for updating and retrieving data by triggering actions. The dispatcher is
+responsible for receiving these actions and directing them to the appropriate stores.
+
+Example:
+
+In the following example we will create a new store for mapping messages to users.
+
+```tsx
+import { Message } from "discord-types/general";
+import { common } from "replugged";
+const { fluxDispatcher: Dispatcher, flux: Flux } = common;
+
+class UserMessagesStore extends Flux.Store {
+  public userMessagesMap: Map<string, Map<string, Message>>;
+
+  public constructor() {
+    super(Dispatcher, {
+      MESSAGE_CREATE: ({ message }) => {
+        /*
+          This function will be called only when a message is received.
+          So, only when the dispatcher received the MESSAGE_CREATE action type.
+        */
+        if (message.state === "SENDING") return;
+
+        const userId = message.author.id;
+        const userMessages = this.userMessagesMap.get(userId) ?? new Map();
+
+        userMessages.set(message.id, message);
+
+        this.userMessagesMap.set(userId, userMessages);
+      },
+      USER_MESSAGES_DELETE: ({ userId }) => {
+        // This is a custom action type we created for this example. It will delete a specific map entry (userId).
+        this.clearUserMessages(userId);
+      },
+    });
+
+    this.userMessagesMap = new Map(); // Here are stored all our users and their messages
+  }
+
+  public clearUserMessages(userId: string): void {
+    // Function to delete a user from userMessagesMap
+    if (this.userMessagesMap.has(userId)) {
+      this.userMessagesMap.delete(userId);
+    }
+  }
+}
+
+export default new UserMessagesStore();
+```
+
+In this example, the action of deleting an entry from the map can be called with the
+[`dispatcher`](common#fluxDispatcher):
+
+```tsx
+// ...
+Dispatcher.dispatch({
+  type: "USER_MESSAGES_DELETE",
+  userId: "USER_ID",
+});
+```
+
+Or with the appropriate function:
+
+```tsx
+// ...
+store.clearUserMessages("USER_ID");
+```
+
+:::tip
+
+You can expose a store by setting its display name and the `__getLocalVars` function.
+
+```tsx
+class UserMessagesStore extends Flux.Store {
+  public userMessagesMap: Map<string, Map<string, Message>>;
+  public displayName: string;
+
+  public constructor() {
+    // ...
+    this.userMessagesMap = new Map();
+    // highlight-next-line
```

Comment:
```suggestion
```

Reply:
Why? I put it to show that the displayName should be set.